pkg/daemon/config.go

```diff
@@ -107,35 +108,42 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
 
 func (output *ptp4lConf) populatePtp4lConf(config *string) error {
 	lines := strings.Split(*config, "\n")
-	var currentSection string
-	output.sections = make(map[string]ptp4lConfSection)
+	var currentSectionName string
+	var currentSection ptp4lConfSection
+	output.sections = make([]ptp4lConfSection, 0)
+	globalIsDefined := false
 
 	for _, line := range lines {
 		if strings.HasPrefix(line, "[") {
-			currentSection = line
+			if currentSectionName != "" {
+				output.sections = append(output.sections, currentSection)
+			}
+			currentSectionName = line
 			currentLine := strings.Split(line, "]")
 
 			if len(currentLine) < 2 {
 				return errors.New("Section missing closing ']'")
 			}
 
-			currentSection = fmt.Sprintf("%s]", currentLine[0])
-			section := ptp4lConfSection{options: map[string]string{}}
-			output.sections[currentSection] = section
-		} else if currentSection != "" {
+			currentSectionName = fmt.Sprintf("%s]", currentLine[0])
+			if currentSectionName == "[global]" {
+				globalIsDefined = true
+			}
+			currentSection = ptp4lConfSection{options: map[string]string{}, sectionName: currentSectionName}
+		} else if currentSectionName != "" {
 			split := strings.IndexByte(line, ' ')
 			if split > 0 {
-				section := output.sections[currentSection]
-				section.options[line[:split]] = line[split:]
-				output.sections[currentSection] = section
+				currentSection.options[line[:split]] = line[split:]
```

Comment:
So the section are added to an array, that way section is guaranteed to be in order but the content are not, correct ?

Reply:
correct.  There is no requirement for the order within a section to be kept.